cmd/utils/flags.go

```diff
@@ -2075,8 +2075,13 @@ func MakeChainDatabase(ctx *cli.Context, stack *node.Node, readonly bool) ethdb.
 func tryMakeReadOnlyDatabase(ctx *cli.Context, stack *node.Node) ethdb.Database {
 	// If datadir doesn't exist we need to open db in write-mode
 	// so database engine can create files.
+	//
+	// There's also a cornercase where the node's database is removed, but it's
+	// ancient store is left intact. Looking at the existence of chaindata thus
+	// is not enough. Since the 'CURRENT' file is present in both leveldb and
+	// pebble, we can check for that as a dirty hack.
 	readonly := true
-	if !common.FileExist(stack.ResolvePath("chaindata")) {
+	if !common.FileExist(filepath.Join(stack.ResolvePath("chaindata"), "CURRENT")) {
```

Comment:
Perhaps we should use https://github.com/ethereum/go-ethereum/blob/master/core/rawdb/database.go

`if rawdb.HasPreexistingDb(stack.ResolvePath("chaindata")) != "" {`